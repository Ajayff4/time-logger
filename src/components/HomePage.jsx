import React, { Component } from 'react'
import { Switch, Route, Link, withRouter } from "react-router-dom"
import Login from './Login'
import Logs from './Logs'
import AddTag from './AddTag'
import Profile from './Profile'
import AddLog from './AddLog'
import Signup from './Signup'
import AddCategory from './AddCategory'
import * as actions from '../actions'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import PageNotFound from './PageNotFound'
import Home from './Home'

class HomePage extends Component {
    static propTypes = {
        isUserLoggedIn: PropTypes.bool.isRequired,
        username: PropTypes.string,
        fullname: PropTypes.string,
        email: PropTypes.string,
        logs: PropTypes.array,
        logout: PropTypes.func.isRequired,
        unsetCookie: PropTypes.func.isRequired
    }

    handleLogout = () => {
        this.props.unsetCookie()
        this.props.logout()
        this.props.history.push('/home')
    }
    render() {
        let cookieVerify = document.cookie.length;
        return (
            <div className="homeComponent">
            <nav id="menu-bar">
                    <center>
                        <ul>
                            {cookieVerify? (
                                <div>
                                    <li key="add-log"><Link to="/add-log"><button>Add Log</button></Link></li>
                                    <li key="logs"><Link to="/logs"><button>Logs</button></Link></li>
                                    <li key="profile"><Link to="/profile"><button>Profile</button></Link></li>
                                    <li><button onClick={this.handleLogout}>Logout</button></li>
                                </div>
                            ) : (
                                <div>
                                    <li key="home"><Link to="/home"><button>Home</button></Link></li>
                                    <li key="login"><Link to="/login"><button>Login</button></Link></li>
                                    <li key="signup"><Link to="/signup"><button>Signup</button></Link></li>
                                </div>
                            )}
                        </ul>
                    </center>
                    <Switch>
                        <Route path="/" exact ><Home /></Route>
                        <Route path="/home" ><Home /></Route>
                        <Route path="/logs">{cookieVerify?<Logs />:<PageNotFound />}</Route>
                        <Route path="/add-log"><AddLog /></Route>
                        <Route path="/add-tag">{cookieVerify?<AddTag />:<PageNotFound />}</Route>
                        <Route path="/add-category">{cookieVerify?<AddCategory />:<PageNotFound />}</Route>
                        <Route path="/profile">{cookieVerify?
                            <Profile username={this.props.username} fullname={this.props.fullname} email={this.props.email} logs={this.props.logs} />:<PageNotFound />}
                        </Route>
                        <Route path="/login"><Login /></Route>
                        <Route path="/signup"><Signup /></Route>
                    </Switch>
            </nav>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        isUserLoggedIn: state.user.loggedIn,
        username: state.user.username,
        fullname: state.user.fullname,
        email: state.user.email,
        logs: state.log.logs
    }
}

const mapDispatchToProps = dispatch => {
    return {
        unsetCookie: () => dispatch(actions.unsetCookie()),
        logout: () => dispatch(actions.logout())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(HomePage))