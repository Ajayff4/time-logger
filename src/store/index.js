import { createStore, applyMiddleware } from 'redux'
import rootReducer from '../reducers/index'
//import logger from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension'
import rootSaga from '../saga'
import createSagaMiddleware from 'redux-saga'

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(sagaMiddleware))
)

sagaMiddleware.run(rootSaga)

export default store