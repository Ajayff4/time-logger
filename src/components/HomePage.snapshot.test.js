import React from 'react'
import renderer from 'react-test-renderer'
import HomePage from './HomePage'
import { Provider } from 'react-redux';
import store from '../store';
import { BrowserRouter as Router } from 'react-router-dom';

describe('HomePage Snapshot Testing...', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(<Provider store={store}><Router><HomePage /></Router></Provider> ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});