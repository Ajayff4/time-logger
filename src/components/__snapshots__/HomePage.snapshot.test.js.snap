// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Snapshot Testing... renders component correctly 1`] = `
<div
  className="homeComponent"
>
  <nav
    id="menu-bar"
  >
    <center>
      <ul>
        <div>
          <li>
            <a
              href="/home"
              onClick={[Function]}
            >
              <button>
                Home
              </button>
            </a>
          </li>
          <li>
            <a
              href="/login"
              onClick={[Function]}
            >
              <button>
                Login
              </button>
            </a>
          </li>
          <li>
            <a
              href="/signup"
              onClick={[Function]}
            >
              <button>
                Signup
              </button>
            </a>
          </li>
        </div>
      </ul>
    </center>
    <div
      className="homeComponent"
    >
      <center>
        <div
          className="homePageText"
        >
          <h2>
            This app is useful for managing time and also useful for self evaluation.
          </h2>
          <p>
            Made By AA077457
          </p>
          <p>
            Thank You!!!
          </p>
        </div>
      </center>
    </div>
  </nav>
</div>
`;
